### ISSUE TITLE
Rename some tools packages IDs as part of the move to msbuild 

### ISSUE THREAD

**Issue by morrisjoe** [dotnet/cli#4243]
**Tracking issue**
Looking very likely that we’ll have to rename some tools packages IDs as part of the move to msbuild. 
As such, customers migrating projects from project.json to MSBUILD will need to change package IDs (along with versions) for these packages. Is there scope to have the migrate experience do this?
Discussion between Damian and Brian....
While renaming the tools during the migration is admittedly a simple feature, there are a few things I’m worried about and which make me extremely hesitant to commit to this as a migration experience.
First thing is how will Migration determine the new versions of these tools? 
What if these new tool implementations have new imports, will migration need to know about this as well? 
What if the tools have moved from tools to tasks, and should now be dependencies?

What if the tool was before invoked via script, and now shouldn’t be, does migration need to know to do special transformations for this special set of tools?
Will the tools be running on the same version of the shared framework as before or on 1.1.0? Does this mean we have to change the app’s version of the shared framework too, which will cause breakage? 
I’m hesitant because of how much semantic knowledge of tools this brings into the migration utility, but pragmatically I recognize the need.
Damian, can you construct a concrete list of which tools need to undergo changes and the full set of changes they would need, so we can figure out the requirements around this?

**Comment by brthor** [comment]
Assigning Damian, Mike as well so we can get a list of the tools and the changes needed.

**Comment by muratg** [comment]
cc @bricelam

**Comment by livarcocc** [comment]
@muratg @mlorbetske the window is closing to get this done.

Can we get help this week to get this list and the versions we should move to?

**Comment by bricelam** [comment]
For EF, this:
```JSON
{
  "dependencies": {
    "Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
     }
  },
  "tools": {
    "Microsoft.EntityFrameworkCore.Tools": "1.0.0-preview2-final"
  }
}
```
Should become:
```XML
<Project>
  <ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools">
      <Version>1.0.0-preview3-final</Version>
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet">
      <Version>1.0.0-preview3-final</Version>
    </DotNetCliToolReference>
  </ItemGroup>
</Project>
```
There is one more step required, but this should be enough for our tool (`dotnet ef`) to walk the user through completing the upgrade process.

**Comment by muratg** [comment]
cc @natemcmaster @NTaylorMullen @prafullbhosale @pranavkm @Eilon

**Comment by NTaylorMullen** [comment]
For Razor tooling this:
```JSON
{
  "dependencies": {
    "Microsoft.AspNetCore.Razor.Tools" /* OR */ "Microsoft.AspNetCore.Razor.Design": {
      "version": "1.0.0-previewX-final",
      "type": "build"
     }
  },
  "tools": {
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-previewX-final"
  }
}
```
Should become:
```XML
<Project>
  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Razor.Design">
      <Version>1.0.0-rc1-final</Version>
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.AspNetCore.Razor.Tools">
      <Version>1.0.0-rc1-final</Version>
    </DotNetCliToolReference>
  </ItemGroup>
</Project>
```

**Comment by prafullbhosale** [comment]
For Scaffolding: 
```
{
  "dependencies": {
    "Microsoft.VisualStudio.Web.CodeGeneration.Tools": {
      "version": "1.0.0-previewX-final",
      "type": "build"
     },
    "Microsoft.VisualStudio.Web.CodeGenerators.Mvc": {
      "version": "1.0.0-previewX-final",
      "type": "build"
     },
  },
  "tools": {
     "Microsoft.VisualStudio.Web.CodeGeneration.Tools": {
       "version": "1.0.0-*",
       "imports": [
         "portable-net451+win8"
       ]
     }
  }
}
```
Should become: 
```
<Project>
  <ItemGroup>
    <PackageReference Include="Microsoft.VisualStudio.Web.CodGeneration.Design">
      <Version>1.0.0-rc1-final</Version>
      <PrivateAssets>All</PrivateAssets>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodGeneration.Tools">
      <Version>1.0.0-rc1-final</Version>
    </DotNetCliToolReference>
  </ItemGroup>
</Project>
```

**Comment by livarcocc** [comment]
@murat So we have EF, Razor and Scaffolding here. Is this all?

**Comment by muratg** [comment]
@livarcocc Should be... although I'd like to get @natemcmaster (OOF today, will be back tomorrow) to also review.

**Comment by natemcmaster** [comment]
Here are two more: "dotnet-watch" and "dotnet-user-secrets" both `1.0.0-preview2-final` and `1.0.0-preview3-final`versions should be upgraded to RC. No change to package ID, just versions.
```json
{
  "tools": {
    "Microsoft.DotNet.Watcher.Tools": "1.0.0-*",
    "Microsoft.Extensions.SecretManager.Tools": "1.0.0-*"
  }
}
```

```xml
<Project>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.DotNet.Watcher.Tools" Version="1.0.0-rc1-final" />
    <DotNetCliToolReference Include="Microsoft.Extensions.SecretManager.Tools" Version="1.0.0-rc1-final" />
  </ItemGroup>
</Project>
```

**Comment by natemcmaster** [comment]
Just thought of another scenario:

dotnet-publish-iis (Microsoft.AspNetCore.Server.IISIntegration.Tools) will be deprecated. This functionality has become a target instead in the web SDK.

cc @pranavkm what is the plan for precompiling razor views (Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools)? Will an MSBuild version be just a target or will it continue to be a CLI tools? (Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools)

**Comment by pranavkm** [comment]
The tool will be converted to a target. The Design package will continue to exist in some format.

**Comment by livarcocc** [comment]
So, we should not migrate Microsoft.AspNetCore.Server.IISIntegration.Tools and Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools. Correct?

**Comment by natemcmaster** [comment]
Right, we should not migrate Microsoft.AspNetCore.Server.IISIntegration.Tools. It's functionality has moved to "**Microsoft.NET.Sdk.Web**". Will migrate install a package ref to the .Web SDK for web projects?

@pranavkm which package is going to contain the target version of Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools?

**Comment by pranavkm** [comment]
The plan was to have Microsoft.AspNetCore.Mvc.Razor to support precompilation out of the box, so presumably we would put it there. I'll run it by @DamianEdwards \ @Eilon

**Comment by livarcocc** [comment]
@natemcmaster Yes, I will add a reference to it when migrating. @balachir just opened a bug asking for that.

@pranavkm Please let me know what to migrate Precompilation.Tools to when you close on it.

**Comment by pranavkm** [comment]
@livarcocc the first version of the tool was released as part of out 1.1.0 preview1 release (and we have plans to change the tool name as part the RTM release (https://github.com/aspnet/MvcPrecompilation/issues/25). I'm don't think we have work to be done here for the `1.0.0-preview3` release of the CLI.

**Comment by pranavkm** [comment]
Spoke to @Eilon about this - we should be fine skipping MvcPrecompilation for this release.
